src/formats/png.zig

```diff
@@ -63,9 +74,137 @@ pub const PNG = struct {
     }
 
     pub fn writeImage(allocator: Allocator, write_stream: *Image.Stream, image: Image, encoder_options: Image.EncoderOptions) ImageWriteError!void {
-        _ = allocator;
-        _ = write_stream;
-        _ = image;
-        _ = encoder_options;
+        const options = encoder_options.png;
+        
+        var writer = write_stream.writer();
+
+        try writeSignature(writer);
+        try writeHeader(writer, image, options);
+        if (image.pixelFormat().isIndex()) {
+            try writePalette(writer, image);
+            try writeTransparencyInfo(writer, image); // TODO: pixel format where there is no transparency
+        }
+        try writeData(allocator, writer, image, options);
+        try writeTrailer(writer);
+    }
+
+    fn writeSignature(writer: anytype) !void {
+        try writer.writeAll("\x89\x50\x4E\x47\x0D\x0A\x1A\x0A");
```

Comment:
There already the magic header in png/types.png: `magic_header` in a more human-readable version